cnb_image.go

```diff
@@ -22,10 +23,13 @@ type CNBImageCore struct {
 	// required
 	v1.Image // the working image
 	// optional
-	createdAt           time.Time
-	preferredMediaTypes MediaTypes
-	preserveHistory     bool
-	previousImage       v1.Image
+	createdAt                    time.Time
+	preferredMediaTypes          MediaTypes
+	preserveHistory              bool
+	previousImage                v1.Image
+	os, arch, variant, osVersion string
+	features, osFeatures, urls   []string
+	annotations                  map[string]string
```

Comment:
I noticed you want to keep a reference to this values to speed up some methods and avoid reading the Config file, but is it really making things faster?, I think the information we try to keep i `CNBImage` is the one we don't have in the underlying `v1.Image`